lib/booking_bug/api.rb

```diff
@@ -0,0 +1,37 @@
+class BookingBug
+  class UnableToAuthenticate < StandardError; end
+
+  class Api
+    def initialize
+      @connection = BookingBug::Connection.new
+    end
+
+    def auth_token
+      @auth_token ||= @connection.login['auth_token']
+    rescue => e
+      raise UnableToAuthenticate, e.message
+    end
+
+    def all
+      bookings = []
+      url = base_path
+
+      while url
+        data = @connection.page(url, auth_token)
+
+        bookings += data['_embedded']['bookings']
+
+        # TODO: replace the below with try once we have active-support!
+        url = data['_links']['next'] && data['_links']['next']['href']
+      end
+
+      bookings
+    end
+
+    private
+
+    def base_path
+      "/api/v1/admin/#{ENV['BOOKING_BUG_COMPANY']}/bookings"
```

Comment:
`BookingBug::Api` doesn't need to decide how it should be configured (in this case using `ENV`). By doing so we've coupled the class to `ENV`.

If we were to introduce a "company" attribute we can remove the coupling. We can set a sensible default to make using the class easier too.

```rb
class BookingBug::API
  def initialize(company_id: ENV['BOOKING_BUG_COMPANY'])
    self.company_id = company_id
  end

  attr_accessor :company_id

  private

  def base_path
    "/api/v1/admin/#{company_id}/bookings"
  end
end
```